iOS_SDK/OneSignalSDK/Source/OSMessagingController.m

```diff
@@ -266,12 +355,30 @@ - (void)handleMessageActionWithURL:(OSInAppMessageAction *)action {
     }
 }
 
+/*
+ * Part of redisplay logic
+ *
+ * Make all messages with redisplay available if:
+ *   - Already displayed
+ *   - At least one Trigger has changed
+ */
+- (void)makeRedisplayedMessagesAvailableWithTriggers:(NSArray<NSString *> *)newTriggersKeys {
+    for (OSInAppMessage *message in _messages) {
+        if ([_redisplayedInAppMessages objectForKey:message.messageId] &&
+            [self.triggerController isTriggerOnMessage:message newTriggersKeys:newTriggersKeys]) {
+              message.isTriggerChanged = true;
+        }
+    }
+}
+
 #pragma mark Trigger Methods
 - (void)addTriggers:(NSDictionary<NSString *, id> *)triggers {
+    [self makeRedisplayedMessagesAvailableWithTriggers:triggers.allKeys];
```

Comment:
Maybe change name to `evaluateRedisplayedInAppMessages`? Little shorter and more direct

Reply:
sure! thats nicer